v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/metadata/CassandraSourceMetadata.java

```diff
@@ -0,0 +1,245 @@
+    Map<String, com.google.cloud.teleport.v2.spanner.migrations.schema.SourceTable> sourceTableMap =
+        convertSourceSchemaToMap(sourceSchema);
+    schema.setSrcSchema(sourceTableMap);
+    schema.setToSource(convertSourceToNameAndColsTable(sourceSchema.tables().values()));
+  }
+
+  /**
+   * Converts a {@link SourceSchema} to a map of Spanner table names to {@link SourceTable} objects.
+   *
+   * @param sourceSchema The SourceSchema to convert.
+   * @return A map where the key is the table name and the value is the corresponding {@link
+   *     SourceTable}.
+   */
+  private Map<String, com.google.cloud.teleport.v2.spanner.migrations.schema.SourceTable>
+      convertSourceSchemaToMap(SourceSchema sourceSchema) {
+    return sourceSchema.tables().entrySet().stream()
+        .collect(
+            Collectors.toMap(
+                Map.Entry::getKey, entry -> convertSourceTableToSourceTable(entry.getValue())));
+  }
+
+  /**
+   * Converts a {@link SourceTable} to a {@link SourceTable} for Spanner.
+   *
+   * @param sourceTable The SourceTable to convert.
+   * @return A converted {@link SourceTable} object suitable for Spanner.
+   */
+  private com.google.cloud.teleport.v2.spanner.migrations.schema.SourceTable
+      convertSourceTableToSourceTable(SourceTable sourceTable) {
+    List<SourceColumn> columns = sourceTable.columns();
+
+    String[] colIds = columns.stream().map(SourceColumn::name).toArray(String[]::new);
+
+    Map<String, SourceColumnDefinition> colDefs =
+        columns.stream()
+            .collect(
+                Collectors.toMap(
+                    SourceColumn::name,
+                    col ->
+                        new SourceColumnDefinition(
+                            col.name(),
+                            new SourceColumnType(col.sourceType(), new Long[0], new Long[0]))));
+
+    ColumnPK[] primaryKeys =
+        columns.stream()
+            .filter(SourceColumn::isPrimaryKey)
+            .map(col -> new ColumnPK(col.name(), getPrimaryKeyOrder(col)))
+            .toArray(ColumnPK[]::new);
+
+    return new com.google.cloud.teleport.v2.spanner.migrations.schema.SourceTable(
+        sourceTable.name(), null, colIds, colDefs, primaryKeys);
+  }
+
+  /**
+   * Determines if a column is a primary key based on its kind.
+   *
+   * @param kind The column kind (e.g., "partition_key" or "clustering").
+   * @return true if the column is a primary key, false otherwise.
+   */
+  private boolean isPrimaryKey(String kind) {
+    return "partition_key".equals(kind) || "clustering".equals(kind);
+  }
+
+  /**
+   * Gets the order of the primary key column.
+   *
+   * @param col The {@link SourceColumn} representing the primary key column.
+   * @return The order of the primary key.
+   */
+  private int getPrimaryKeyOrder(SourceColumn col) {
+    switch (col.kind()) {
+      case "partition_key":
+        return 1;
+      case "clustering":
+        return 2;
```

Comment:
What will happen in case of composite clustering keys?

Reply:
composite clustering keys? -> as per the analysis I guess in Bith case kind come as **clustering** only

@shreyakhajanchi as we are not using order any where so it does not create any issue between yes we are handing both the key marking as primary key duing DML

Reply:
In that case you can always set 1 and remove this method

Reply:
ok @shreyakhajanchi